Python/Product/TestAdapter.Executor/Common/PythonTestDiscoverer.cs

```diff
@@ -58,6 +34,13 @@ public abstract class PythonTestDiscoverer : ITestDiscoverer {
             _frameworkType = frameworkType;
         }
 
+        abstract public void DiscoverTests(
+           IEnumerable<string> sources,
+           PythonProjectSettings settings,
+           IMessageLogger logger,
+           ITestCaseDiscoverySink discoverySink
+        );
+
```

Comment:
New abstract method that does the actual work.

Note: When running from command line the test executor will also call this, thats why its public